api/api_subsystems.go

```diff
@@ -0,0 +1,63 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+type MinerSubsystems []MinerSubsystem
+
+func (ms MinerSubsystems) Has(entry MinerSubsystem) bool {
+	for _, v := range ms {
+		if v == entry {
+			return true
+		}
+
+	}
+	return false
+}
+
+type MinerSubsystem int
+
+const (
+	MarketsSubsystem MinerSubsystem = iota
+	MiningSubsystem
+	SealingSubsystem
+	SectorStorageSubsystem
+)
+
+func (ms MinerSubsystem) String() string {
+	return MinerSubsystemToString[ms]
+}
+
+var MinerSubsystemToString = map[MinerSubsystem]string{
+	MarketsSubsystem:       "Markets",
+	MiningSubsystem:        "Mining",
+	SealingSubsystem:       "Sealing",
+	SectorStorageSubsystem: "SectorStorage",
+}
```

Comment:
Doesn't hurt too much in this case, but we should use a static array since the underlying type is an int, thus avoiding a map lookup every time we want to return the string representation.